File: src/contests/cg/puzzle/dart/sum-of-divisors.dart
// 🎮 CodinGame Puzzle - sum-of-divisors
// https://www.codingame.com/training/medium/sum-of-divisors

import 'dart:io';

void main() {
  // d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
  // of divisors is sum over d of d * floor(n/d).
  int n = int.parse(stdin.readLineSync()!.trim());
  int total = 0;
  for (int d = 1; d <= n; d++) total += d * (n ~/ d);
  print(total);
}

File: src/contests/cg/puzzle/rust/sum-of-divisors.rs
// 🎮 CodinGame Puzzle - sum-of-divisors
// https://www.codingame.com/training/medium/sum-of-divisors

fn rl() -> String {
    let mut s = String::new();
    std::io::stdin().read_line(&mut s).unwrap();
    s.trim().to_string()
}

fn main() {
    // d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
    // of divisors is sum over d of d * floor(n/d).
    let n: i64 = rl().parse().unwrap();
    let mut total: i64 = 0;
    let mut d: i64 = 1;
    while d <= n {
        total += d * (n / d);
        d += 1;
    }
    println!("{}", total);
}
